api/v1/design/resource.go

```diff
@@ -280,4 +280,35 @@ var _ = Service("resource", func() {
 		})
 	})
 
+	Method("GetYamlByCatalogKindNameVersion", func() {
+		Description("Fetch a resource yaml file using the name of catalog, resource name, kind, and version")
+
+		Payload(func() {
+			Attribute("catalog", String, "name of catalog", func() {
+				Example("catalog", "tektoncd")
+			})
+			Attribute("kind", String, "kind of resource", func() {
+				Enum("task", "pipeline")
+			})
+			Attribute("name", String, "name of resource", func() {
+				Example("name", "buildah")
+			})
+			Attribute("version", String, "version of resource", func() {
+				Example("version", "0.1")
+			})
+
+			Required("catalog", "kind", "name", "version")
+		})
+
+		Result(String)
+
+		HTTP(func() {
+			GET("/{catalog}/{kind}/{name}/{version}")
```

Comment:
I think we might have to change the endpoint as we already have this endpoint which returns the resource details for the specific version

- For e.g. https://api.hub.tekton.dev/v1/resource/tekton/task/buildah/0.1 

Also we have an endpoint as `/{catalog}/{kind}/{name}/{version}/yaml` which returns the yaml of the resource but in the format of json

```json
{
   "data": {
        "yaml": "This works"
    }
}
```

- For example: https://api.hub.tekton.dev/v1/resource/tekton/task/buildah/0.1/yaml

  - The implementation for the above can be found [here](https://github.com/tektoncd/hub/pull/539/files#diff-614f7626d1fed615086cc2f5c018905d774707638e765a91ca9e180b8636529cR202) and may be we can reuse it

So we can have two solutions for this 
- Either update the current endpoint /{catalog}/{kind}/{name}/{version}/yaml` to return a string and correspondingly update the ui logic
- Secondly have this endpoint with a new name 

And also based on the pr description I see we are aiming for `kubectl apply -f https://hub.tekton.dev/tekton/task/buildpacks/0.4` but I think we should use the API and not the UI and also currently `https://hub.tekton.dev/tekton/task/buildpacks/0.4` will open up the description page on Hub UI for buildpacks

Reply:
cc @vinamra28  @jerop

Reply:
thanks the check-in and feedback! 

you are right. there is a quite similar endpoint existing under the service `resources`. the major diff will be the return val is expected as a yaml file, so able to be applied directly. 

perhaps one concern for updating the existing endpoint is the "breaking change" as someone could be relying on this existing endpoint to populate their resources.

Reply:
yes that would be a breaking change and I think that would not be a great solution

We can have an endpoint something like this `/{catalog}/{kind}/{name}/{version}/{name}.yaml` to serve the purpose of this api service

Reply:
yap. good point. was thinking the original TEP want to let the URL/endpoint get rid of the specific file name, so only locating the resources mainly with the resource name + version under a certain catalog.

Hi @jerop / @vinamra28 any thoughts for this thread?

Reply:
hey @austinzhao-go, yeah I agree that the TEP wants to get rid of specific file name from the URL/endpoint, how about we try to include `raw` keyword in the URL(I don't have any strong opinions for the appropriate location) one such suggestion can be - `/resource/raw/{catalog}/{kind}/{name}/{version}/yaml`. This way we can avoid adding file name in the URL.

In https://github.com/tektoncd/community/pull/663#discussion_r884802165, I also proposed the same, what @PuneetPunamiya suggested, keeping in mind the current implementation of github, how it provides the raw URL (eg https://raw.githubusercontent.com/tektoncd/catalog/main/task/ansible-tower-cli/0.1/ansible-tower-cli.yaml) but yeah I don't have strong opinions. Probably we can discuss in one of the WG calls so that we all can come on a conclusion. What do you think?

/cc @tektoncd/hub-maintainers @piyush-garg @jerop

Reply:
yap! thanks a lot for offering me more details and related work! 

will bring for a check in next WG. and welcome further feedback for this TEP and Impl in this thread.

Reply:
I'm open to any path the hub maintainers would prefer to use, just as long as it serves the yaml that users can directly apply -- happy to update the TEP as needed. Thank you @austinzhao-go @PuneetPunamiya @vinamra28!

Reply:
discussed with @vinamra28 and agreed that we go with @PuneetPunamiya's suggestion to have `/{catalog}/{kind}/{name}/{version}/{name}.yaml`

for example https://hub.tekton.dev/tekton/task/buildpacks/0.4/buildpacks.yaml

cc @austinzhao-go

Reply:
cool. thanks @jerop

will keep this new endpoint under the `resource` service.